src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultPlugin.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.KeyVault
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.ServiceBus.Core;
+    using System.Security.Cryptography;
+    using System.IO;
+    using System.Text;
+
+    /// <summary>
+    /// Provides Azure KeyVault functionality for Azure Service Bus.
+    /// </summary>
+    public class KeyVaultPlugin : ServiceBusPlugin
+    {
+        private readonly string secretName;
+        private readonly string keyVaultEndpoint;
+        private KeyVaultSecretManager secretManager;
+
+        /// <summary>
+        /// Gets the name that is used to identify this plugin.
+        /// </summary>
+        public override string Name => "KeyVaultPlugin";
```

Comment:
I'd call it "Microsoft.Azure.ServiceBus.KeyVault.KeyVaultPlugin", but it's just me 😉